services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/MongoDbBasedTenantBackend.java

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.deviceregistry.mongodb;
+
+import java.util.Optional;
+
+import javax.security.auth.x500.X500Principal;
+
+import org.eclipse.hono.deviceregistry.mongodb.service.MongoDbBasedTenantService;
+import org.eclipse.hono.service.management.Id;
+import org.eclipse.hono.service.management.OperationResult;
+import org.eclipse.hono.service.management.Result;
+import org.eclipse.hono.service.management.tenant.Tenant;
+import org.eclipse.hono.service.management.tenant.TenantBackend;
+import org.eclipse.hono.util.TenantResult;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Repository;
+
+import io.opentracing.Span;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.Verticle;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * A tenant backend for a mongodb based of device registry.
+ */
+@Repository
+@Qualifier("backend")
+@ConditionalOnProperty(name = "hono.app.type", havingValue = "mongodb", matchIfMissing = true)
+public class MongoDbBasedTenantBackend extends AbstractVerticle
```

Comment:
Could you pls place in an appropriate package? The device counterpart namely `MongoDbBasedDeviceBackend` resides under `org.eclipse.hono.deviceregistry.mongodb.service`.